codegen/src/main/java/software/amazon/awssdk/codegen/poet/auth/scheme/AuthSchemeInterceptorSpec.java

```diff
@@ -255,13 +262,40 @@ private MethodSpec generateTrySelectAuthScheme() {
                              ResolveIdentityRequest.class,
                              ResolveIdentityRequest.class);
         builder.addStatement("authOption.forEachIdentityProperty(identityRequestBuilder::putProperty)");
-        builder.addStatement("$T identity = identityProvider.resolveIdentity(identityRequestBuilder.build())",
-                             namedIdentityFuture());
+
+        builder.addStatement("$T<? extends T> identity", CompletableFuture.class);
```

Comment:
nit: May want to use a combination of `ParameterizedTypeName` and `TypeVariableName` to be more idiomatic

Reply:
ah, good catch. `namedIdentityFuture()` was already there in previous code. I just moved it incorrectly to line 270 below in this change. fixed.